deploy/infrastructure/main.tf

```diff
@@ -27,3 +27,19 @@ module "iam_eks_role" {
     # s3 = aws_iam_policy.read_s3.arn
   }
 }
+
+# Created in the platform by default. Overridable
+data "aws_route53_zone" "this" {
+  name = var.dns_zone_domain
+}
+
+# Certificate is picked automatically by the ALB Controller through auto-discovery
+module "acm" {
```

Comment:
will not work if the DNS zone is in cloudflare no ?

Reply:
@rbapst-tamedia It won't work. You are correct. Let me explain why I left it out at the moment.

Adding CloudFlare means we need to have instantiated the CloudFlare provider. At the moment the Tam CLI features don't support removing/adding blocks of code (if we don't have cloudflare tokens the terraform init will fail). For that either we:
- build the feature in the Tam CLI
- (or) use one of the many existing template engines like [`jinja2`](https://palletsprojects.com/p/jinja/) or [`cookiecutter`](https://github.com/cookiecutter/cookiecutter)

So, I was thinking of including only Route53 and to create a separate ticket for CloudFlare + template engine. What do you think?

I think it's time to step up the game of the Tam CLI with more complex templating features, not just token replacement.

Reply:
That's fine, only issue is that you implicitly need a route53 zone for your app. So workflow will fail too.
At least I would put a condition.

Reply:
@rbapst-tamedia I think it's ok to have it fail at the moment if you don't have a Route53 zone. Since it requires anyway manual action to setup the certificate, you would need to touch that code as well. 

(I think it's good to avoid complexity in an intermediate state as is now)

Reply:
Ok to merge. And it will work if it's the tam-cli which decide the value of var.hostname and var.dns_zone_name depending on 
- team
- destination cluster
- application name
the developer gives when running the tam-cli.
----

The problem is still there if for example on prod, the app should be in a tenant zone (ex: ai-tool.tagesanzeiger.ch) It's means that we do have to create the record for the certificate in cloudflare. no other option.

next steps are:
- Have a route53 zone in each AWS account and a way to identify it
- Find a way to make an application reachable from another DNS name.

Reply:
@rbapst-tamedia You are right in saying that it's a bit more complex and that there are edge cases.

I think the goal should be to:
1. Provide a framework to pre-fill values
2. Allow for full customizability (e.g. specifying the name as input or in the config YAML)
3. Provide a default based on the team

Little by little we will reach that 👍